### PR TITLE
Added store unit sets

### PR DESCRIPTION
**Pull request by vgarvardt**

### CODE REVIEW

File: pkg/store/redis.go [hunk 1]
```diff
@@ -22,6 +22,9 @@ type RedisStore struct {
 
 	// The maximum number of retry under race conditions.
 	MaxRetry int
+
+	// Redis connection stub for tests
+	mockGetConnection func() redis.Conn
```
Comment (italolelis) [inline pkg/store/redis.go]: This really doesn't feel right... having a mocking method on a struct.
Probably better to have an `TestingStore` where you can mock the `GetConnection` method

**Review by italolelis — CHANGES_REQUESTED** [review]

File: pkg/store/redis.go (hunk 1 above)
Comment (vgarvardt) [reply pkg/store/redis.go]: Removed mock method and extracted all logic to private methods that are tested now.

**Comment by vgarvardt** [comment]
Updated PR to reflect the latest changes from https://github.com/hellofresh/janus/pull/85

**Review by italolelis — APPROVED** [review]

**Comment by italolelis** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/hellofresh/api-gateway/pull-request/71/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=hellofresh/api-gateway)

**Pull request merged**